### ISSUE TITLE
doesn't work with image tomcat:9

### ISSUE THREAD

**Issue by ozbillwang** [DataDog/dd-agent#3893]
```
$ cat Dockerfile

FROM tomcat:9

ARG DATADOG_LICENSE="14xxxxxxxx13d3b148727f16721"

# Install host agent
RUN DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=${DATADOG_LICENSE} DD_SITE="datadoghq.com" bash -xc "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"
```
when run the build, I got the error
```
* Starting the Agent...

+ eval ' service datadog-agent restart'
++ service datadog-agent restart
Restarting Datadog Agent
Stopping Datadog Agent: datadog-agent.
Stopping Datadog Process Agent: datadog-agent-process.
Stopping Datadog Trace Agent (APM): datadog-agent-trace.
Starting Datadog Agent: datadog-agentstart-stop-daemon: matching only on non-root pidfile /opt/datadog-agent/run/agent.pid is insecure
start-stop-daemon: matching only on non-root pidfile /opt/datadog-agent/run/agent.pid is insecure
start-stop-daemon: matching only on non-root pidfile /opt/datadog-agent/run/agent.pid is insecure
 failed!

It looks like you hit an issue when trying to install the Agent.

Troubleshooting and basic usage information for the Agent are available at:

    https://docs.datadoghq.com/agent/basic_agent_usage/

If you're still having problems, please send an email to support@datadoghq.com
with the contents of ddagent-install.log and we'll do our very best to help you
solve your problem.

The command '/bin/sh -c DD_AGENT_MAJOR_VERSION=7 DD_API_KEY=${DATADOG_LICENSE} DD_SITE="datadoghq.com" bash -c "$(curl -L https://s3.amazonaws.com/dd-agent/scripts/install_script.sh)"' returned a non-zero code: 1
```

**Comment by ozbillwang** [comment]
do a deep debug, I found the command `service datadog-agent stop` can't clean dd-agent at all

```
root@ca2bf4b588d5:/var/log/datadog# service datadog-agent stop
[ ok ] Stopping Datadog Agent: datadog-agent.
[ ok ] Stopping Datadog Process Agent: datadog-agent-process.
[ ok ] Stopping Datadog Trace Agent (APM): datadog-agent-trace.
root@ca2bf4b588d5:/var/log/datadog# ps aux |more
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0   5752  3692 pts/0    Ss   11:19   0:00 bash
dd-agent  3985  1.1  1.1 1410296 97432 ?       Sl   11:35   0:03 /opt/datadog-agent/bin/agent/agent run -p /opt/datadog-agent/run/agent.pid
root      5080  0.0  0.0   9392  3056 pts/0    R+   11:39   0:00 ps aux
root      5081  0.0  0.0   4252   892 pts/0    S+   11:39   0:00 more
```

I have to run with `pkill -15 agent` to clean it nicely.

**Comment by olivielpeau** [comment]
Hi @ozbillwang, sorry for the late reply, this issue was fixed in https://github.com/DataDog/datadog-agent/pull/5957, which was released with Agent v`7.22.0`/`6.22.0`.

A couple of additional notes:
* To build a custom image based on `tomcat` with the agent in it, you can also consider building a [multi-stage docker image](https://docs.docker.com/develop/develop-images/multistage-build/) using the official `datadog/agent` image. Although the recommended deployment would be to run the agent as a separate `datadog/agent` container.
* For future reference, please note that this repository is home to the (older) Agent v5, the repository of the newer Agent v6/v7 you're referring to in this issue is located here: https://github.com/DataDog/datadog-agent/

In any case, this should be fixed now, so I'll go ahead and close this issue. Feel free to open an issue https://github.com/DataDog/datadog-agent or contact our support team if you're facing further issues.

**Comment by ozbillwang** [comment]
Thanks a lot for the updates.
